packages/nx/src/project-graph/project-graph-builder.ts

```diff
@@ -385,6 +389,24 @@ export class ProjectGraphBuilder {
         }
       }
     }
+    // cleanup stale dependencies
+    for (const [targetProject, types] of alreadySetTargetProjects.entries()) {
+      const node = this.graph.nodes[sourceProject];
+      if (node) {
+        for (const depType of types.keys()) {
+          if (
+            depType !== 'implicit' &&
+            !node.data.files.some((f) =>
```

Comment:
All explicit dependencies should be pruned away if there's no file to which they belong anymore.